draft-smyslov-ipsecme-ikev2-mceliece-00.xml

```diff
@@ -130,20 +133,48 @@
+          short payload format. It is not required to always use the big payload format to
+          use Classic McEliece.</t>
+
           <section anchor="child_sa" title="Using Classic McEliece for Child SAs">
-            <t> The Classic McEliece KEM can also be used in the CREATE_CHILD_SA exchange to create
-              IKE and IPsec SAs. That format and size are just the sane, so doing this also requires
-              support for big payloads and for IKE fragmentation.</t>
+            <t> The Classic McEliece KEM can also be used in the CREATE_CHILD_SA exchange (or in the IKE_FOLLOWUP_KE exchange 
+            as defined in <xref target="RFC9370"/>) to rekey
+              IKE SA and to create/rekey IPsec SAs. The format and size are just the same, so doing this also requires
+              support for big payloads and for IKE fragmentation. However, if Classic McEliece 
+              is only used in these exchanges, there is no need to negotiate the IKE_INTERMEDIATE exchange.</t>
           </section>  
         </section>
 
+        <section anchor="transport" title="Transport Considerations">
+            <t> While an IKE SA that utilizes Classic McEliece can run over UDP, in most settings 
+            this will be problematic. The reason is that with tipical MTU size of 1500 bytes, 
+            an IKE message containing Classic McEliece public key will be splitted into several
+            hundreds IKE fragment messages, which will be sent to the network at once.
+            This may lead to network congestion causing loss of some messages, in which case all
+            the fragment messages will be re-sent again (after timeout), since IKE fragment messages are not 
+            individually acknowledged. This will also lead to congestion and the process will repeat
+            until all the fragment messages will be received, but this might not happen.
+            </t>
+
+            <t> For this reason peers may want to use TCP as a transport. Peers may choose between 
+            using TCP for both IKE and ESP as specified in <xref target="RFC9329"/> or using separate 
+            transports for them <xref target="I-D.smyslov-ipsecme-ikev2-reliable-transport"/>.
+            The latter allows to avoid performance degradation when tunnelling TCP trafic 
+            and thus is <bcp14>RECOMMENDED</bcp14>.
+            </t>
+
+            <t>Note, that despite the fact, that TCP can tranfer messages of any size without
+            having problems with IP fragmentation, in case of Classic McEliece peers
+            still have to negotiate IKE fragmentation <xref target="RFC7383" /> as defined
+            in <xref target="mceliece-ikev2" />, even when they choose to use TCP as a transport for IKE.
+            This is due to the limitation imposed by <xref target="RFC9329"/> that individual
+            IKE message in TCP stream cannot exceed 64 Kbytes. Thus, larger IKE messages

```

Comment:
That's a shame. Especially considering that the IKE message itself always had a 32-bit length field.